### PR TITLE
fix(es/fixer): Don't change default decls to default expr exports

### LINKED ISSUE #7584: SimplifyExpr & Fixer should skip ParenExpr at ExportDefaultExpr
### Describe the bug

SimplifyExpr & Fixer should skip ParenExpr at ExportDefaultExpr

### Input code

```typescript
import someFunction from './b';

export default (function someFunction () {
});
```

### Expected behavior

Transform by SimplifyExpr & Fixer should be the origin code.

``` js
import someFunction from './b';

export default (function someFunction () {
});
```

### Actual behavior

But print the below code, the semantics has changed.
``` js
import someFunction from './b';

export default function someFunction () {
};
```

### Version

master code

### PR DESCRIPTION
**Pull request by underfin**
**Description:**

**Related issue:**

 - Closes #7584.

### CODE REVIEW

File: crates/swc_ecma_transforms_optimization/src/simplify/expr/tests.rs [hunk 1]
```diff
@@ -1544,3 +1544,8 @@ fn test_es6_features() {
         "function foo() {return `${false}`}",
     );
 }
+
+#[test]
+fn test_export_default_paren_expr() {
+    fold_same("import fn from './b'; export default (function fn () {});");
```
Comment (magic-akari) [inline crates/swc_ecma_transforms_optimization/src/simplify/expr/tests.rs:1550]: Another case should be added

```JavaScript
export default ((foo));
```

and expected output:

```JavaScript
export default foo;
```

**Review by magic-akari — CHANGES_REQUESTED** [review]

File: crates/swc_ecma_transforms_optimization/src/simplify/expr/tests.rs (hunk 1 above)
Comment (underfin) [reply crates/swc_ecma_transforms_optimization/src/simplify/expr/tests.rs:1550]: @magic-akari i resolved it, please review it again.

File: crates/swc_ecma_transforms_base/src/fixer.rs [hunk 2]
```diff
@@ -438,6 +438,12 @@ impl VisitMut for Fixer<'_> {
     }
 
     fn visit_mut_export_default_expr(&mut self, node: &mut ExportDefaultExpr) {
+        if let Expr::Paren(p) = &mut *node.expr {
+            if let Expr::Fn(f) = &mut *p.expr {
```
Comment (kdy1) [inline crates/swc_ecma_transforms_base/src/fixer.rs:442]: I thnk we need to check for `Expr::Class`, too

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thanks!

---

swc-bump:
 - swc_ecma_transforms_base

File: crates/swc_ecma_transforms_base/src/fixer.rs (hunk 2 above)
Comment (underfin) [reply crates/swc_ecma_transforms_base/src/fixer.rs]: Thanks a lot !

**Review by kdy1 — APPROVED** [review]
Thanks!

**Comment by underfin** [comment]
@kdy1 I don't know update all snapshot with test, can you point me out? Thanks !

**Comment by kdy1** [comment]
`UPDATE=1 cargo test` will do it, but it does not modify rust files.

**Comment by underfin** [comment]
Thanks! I'm so sad because i want to update the snapshot at rust files test ...

**Pull request merged**